core/drivers/pl022_spi.c

```diff
@@ -212,11 +210,9 @@ static enum spi_result pl022_txrx16(struct spi_chip *chip, uint16_t *wdat,
 	/* Capture remaining rdat not read above */
 	if (rdat) {
 		while ((j < num_pkts) &&
-			(io_read8(pd->base + SSPSR) & SSPSR_BSY))
-			if (io_read8(pd->base + SSPSR) & SSPSR_RNE) {
-				/* rx 1 packet */
-				rdat[j++] = io_read8(pd->base + SSPDR);
-			}
+			(io_read8(pd->base + SSPSR) & SSPSR_RNE))
+			/* rx 1 packet */
+			rdat[j++] = io_read16(pd->base + SSPDR);
```

Comment:
Nitpicking: propose to move the fix `io_read8()`->`io_read16()` from commit "pl022_spi: simplify receive of remaining data" to commit "pl022_spi: fix non-trivial typo".

Reply:
Hrm, that was the way it was done originally. Must have messed up the rebase. Anyway, fixed.